### PR TITLE
Object namespace has higher priority when creating

### LINKED ISSUE #204: Explicitly set namespace in a resource must have higher priority then setting RequestContext
This problem is explained in #201 - level 2 

This is related #202 but not fixed by #203 

In case when we set the namespace for new resource explicitly we will still get 
```
Future(Failure(skuber.api.client.package$K8SException: Status(v1,Status,ListMeta(,),Some(Failure),Some(the namespace of the provided object does not match the namespace sent on the request),Some(BadRequest),None,Some(400))))
```
when requestContext's namespace is different.
The solution will be to teach RequestContext to recognize when namespace is explicitly set and use that namespace to pass it to `buildRequest`
https://github.com/doriordan/skuber/blob/master/client/src/main/scala/skuber/api/package.scala#L242

### PR DESCRIPTION
**Pull request by ngortheone**
* Fix #204
* Some docs
* Some formatting

### CODE REVIEW

File: client/src/main/scala/skuber/api/package.scala [hunk 1]
```diff
@@ -376,19 +376,32 @@ package object client {
...
+         case _                => Some(obj.name)
        }
        modify(method, obj, nameComponent)
      }
 
      private[skuber] def  modify[O <: ObjectResource](method: HttpMethod, obj: O, nameComponent: Option[String])(
        implicit fmt: Format[O], rd: ResourceDefinition[O], lc: LoggingContext): Future[O] =
      {
+
+       // Object namespace has higher priority.
```
Comment (doriordan) [inline client/src/main/scala/skuber/api/package.scala]: In this case I think the code is fairly straightforward and explained in the issue, so I think something like the following comment is enough:
```
      // Namespace set in the object metadata (if set) has higher priority than that of the
      //  request context (see Issue #204)
       val targetNamespace = if (obj.metadata.namespace.isEmpty) namespaceName else obj.metadata.namespace
```

File: client/src/it/scala/skuber/NamespaceSpec.scala [hunk 2]
```diff
@@ -13,27 +15,49 @@ import scala.util.{Failure, Success}
   */
 class NamespaceSpec extends K8SFixture with Eventually with Matchers {
 
-  val nginxPodName: String = java.util.UUID.randomUUID().toString
+  val nginxPodName1: String = randomUUID().toString
+  val nginxPodName2: String = randomUUID().toString
+
+  val namespace1Name: String = "namespace1"
+  val namespace2Name: String = "namespace2"
+  val testNamepsaces = List(namespace1Name,namespace2Name)
```
Comment (doriordan) [inline client/src/it/scala/skuber/NamespaceSpec.scala]: type: should be `testNamespaces`

**Comment by doriordan** [comment]
@ngortheone I've requested just a few changes with respect to comments and typos, otherwise I think this looks good to go. Thanks!

**Comment by doriordan** [comment]
(I am also assuming that the modified NamespaceSpec integration test has been successfully run).

File: docs/Examples.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # Skuber usage examples
 
+Skuber is built in top of Akka HTTP and therefore it has non-blocking and concurrent by default.
```
[the remark this answers is not in the document]
Comment (ngortheone) [reply docs/Examples.md]: Thanks, English is not my native language.

File: client/src/main/scala/skuber/api/package.scala (hunk 1 above)
Comment (ngortheone) [reply client/src/main/scala/skuber/api/package.scala]: ack

File: client/src/it/scala/skuber/NamespaceSpec.scala (hunk 2 above)
Comment (ngortheone) [reply client/src/it/scala/skuber/NamespaceSpec.scala]: ack

**Comment by ngortheone** [comment]
@doriordan I have addressed your comments.
I did run tests in NamespaceSpec many times before submitting this PR - they run successfully

**Comment by doriordan** [comment]
Thanks LGTM

**Pull request merged**